### PR TITLE
Add note on reset! usage in recurrence docs

### PR DESCRIPTION
**Pull request by jpsamaroo**
Closes https://github.com/FluxML/Flux.jl/issues/640

### CODE REVIEW

File: docs/src/models/recurrence.md [hunk 1]
```diff
@@ -114,3 +114,13 @@ truncate!(m)
 Calling `truncate!` wipes the slate clean, so we can call the model with more inputs without building up an expensive gradient computation.
 
 `truncate!` makes sense when you are working with multiple chunks of a large sequence, but we may also want to work with a set of independent sequences. In this case the hidden state should be completely reset to its original value, throwing away any accumulated information. `reset!` does this for you.
+
+Note that when training with recurrent layers in your model, `reset!` needs to be called for each loss calculation:
```
Comment (MikeInnes) [inline docs/src/models/recurrence.md]: I think something like `In general, you'll want to...` since it's not strictly required depending on what behaviour you want.

Also, you can use either `truncate!` or `reset!`.

Comment (jpsamaroo) [reply docs/src/models/recurrence.md]: In what cases can `reset!`/`truncate!` be avoided in the loss calculation? I wasn't able to figure out an alternative with my LSTM example.

Comment (MikeInnes) [reply docs/src/models/recurrence.md]: You wouldn't usually want to for something like a simple LSTM. For something like char-rnn you could in theory backprop over all input the model has seen; things are also a little more complex with things like moving window BPTT.

Of course ideally we'd document all these patterns, but with all the APIs about to shift I think it's ok to wait until Zygote is in to be really thorough here.

Comment (jpsamaroo) [reply docs/src/models/recurrence.md]: PR updated

**Comment by MikeInnes** [comment]
:+1: Thanks!
